src/landing_game_object.py

```diff
@@ -50,3 +52,6 @@ def set_pos(self, pos_pixel: Vec2d) -> None:
 
     def add_pos_to_dict(self):
         self.__dict__.update({"pos": self.pos})
+
+    def __del__(self):
+        del self
```

Comment:
Interesting.  Does this custom destructor  bring benefits?

Reply:
In my first version, it made sure to delete unused IDs. Left it there because I wanted to check if it is also needed in this version. Didn't have enough time yet.